go/oasis-node/cmd/common/common.go

```diff
@@ -200,3 +202,15 @@ func LoadEntity(dataDir string) (*entity.Entity, signature.Signer, error) {
 	entitySignerFactory := fileSigner.NewFactory(dataDir, signature.SignerEntity)
 	return entity.Load(dataDir, entitySignerFactory)
 }
+
+func LoadSigner(signer string) (signature.Signer, error) {
+	switch signer {
+	case "ledger":
+		return ledgerSigner.NewSigner()
+	case "file":
+		_, signer, err := LoadEntity(flags.FileSignerDir())
```

Comment:
This is incorrect as if `signer.file.dir` is different from the data directory where the entity descriptor is kept, entity loading will fail.

When you move this to `LoadEntity`, you should differentiate between the data directory (should be used as the `entity.Load` argument) and the signer directory (should be used as the `fileSigner.NewFactory` argument).